### ISSUE TITLE
Machine Learning 10-701/15-781: Lectures

### ISSUE THREAD

**Issue by guevara** [guevara/read-it-later#6419]
Machine Learning 10-701/15-781: Lectures<br>
<br>
https://ift.tt/udO8dH<br>
<br>
About 0 Minutes<br>
<br>
<div>
<div>
<br>
<br>
<br>
<br>
<br>
<br>
via www.cs.cmu.edu http://www.cs.cmu.edu<br>
<br>
May 2, 2020 at 09:21PM
<table>
<tr>
<th>Date</th>
<th>Lecture</th>
<th>Topics</th>
<th>Readings and useful links</th>
<th>Handouts</th>
</tr>
<tr>
<td>Jan 11</td>
<td>Intro to ML<br>
Decision Trees<br>
<a href="http://www.cs.cmu.edu/~tom/10701_sp11/slides/DTreesAndOverfitting-1-11-2011_final.pdf">Slides</a>
<p><a href="http://cc-web.isri.cmu.edu/CourseCast/Viewer/Default.aspx?id=257476ca-e66a-4cdb-9ffc-8abdd3129954">video</a></p>
</td>
<td>
<ul>
<li>Machine learning examples</li>
<li>Well defined machine learning problem</li>
<li>Decision tree learning</li>
</ul>
</td>
<td>Mitchell: Ch 3<br>
Bishop: Ch 14.4<br>
<a href="http://www.cs.cmu.edu/%7Etom/pubs/MachineLearning.pdf">The Discipline of Machine Learning</a>
</td>
<td>&nbsp;</td>
</tr>
<tr>
<td>Jan 13</td>
<td>Decision Tree learning
<p>Review of Probability</p>
<p><a href="http://www.cs.cmu.edu/~tom/10701_sp11/slides/Overfitting_ProbReview-1-13-2011-ann.pdf">Annotated slides</a></p>
<p><a href="http://cc-web.isri.cmu.edu/CourseCast/Viewer/Default.aspx?id=ad6574d4-b695-4940-8449-2e09096ed16c">video</a></p>
</td>
<td>
<ul>
<li>The big picture</li>
<li>Overfitting</li>
</ul>
<ul>
<li>Random variables, probabilities</li>
</ul>
</td>
<td>
<a href="http://www.autonlab.org/tutorials/prob18.pdf">Andrew Moore's Basic Probability Tutorial</a><br>
Bishop: Ch. 1 thru 1.2.3<br>
Bishop: Ch 2 thru 2.2</td>
<td>
<a href="http://www.cs.cmu.edu/~tom/10701_sp11/hws.shtml">HW1</a> out Jan 14</td>
</tr>
<tr>
<td>Jan 18</td>
<td></td>
<td>
<ul>
<li>Bayes rule</li>
<li>MLE</li>
<li>MAP</li>
</ul>
</td>
<td>
<a href="http://www.autonlab.org/tutorials/prob18.pdf">Andrew Moore's Basic Probability Tutorial</a><br>
Bishop: Ch. 1 thru 1.2.3<br>
Bishop: Ch 2 thru 2.2</td>
<td></td>
</tr>
<tr>
<td>Jan 20</td>
<td></td>
<td>
<ul>
<li>Conditional independence</li>
<li>Multinomial Naive Bayes</li>
</ul>
</td>
<td>Mitchell: <a href="http://www.cs.cmu.edu/%7Etom/mlbook/NBayesLogReg.pdf">Naive Bayes and Logistic Regression</a>
</td>
<td></td>
</tr>
<tr>
<td>Jan 25</td>
<td></td>
<td>
<ul>
<li>Gaussian Bayes classifiers</li>
<li>Document classification</li>
<li>Brain image classification</li>
<li>Form of decision surfaces</li>
</ul>
</td>
<td>Mitchell: <a href="http://www.cs.cmu.edu/%7Etom/mlbook/NBayesLogReg.pdf">Naive Bayes and Logistic Regression</a>
</td>
<td>
<p>HW1 due<br>
HW2 out</p>
</td>
</tr>
<tr>
<td>Jan 27</td>
<td></td>
<td>
<ul>
<li>Naive Bayes - the big picture</li>
<li>Logistic Regression: Maximizing conditional likelihood</li>
<li>Gradient ascent as a general learning/optimization method</li>
</ul>
</td>
<td>Mitchell: <a href="http://www.cs.cmu.edu/%7Etom/mlbook/NBayesLogReg.pdf">Naive Bayes and Logistic Regression</a>
<p>Ng &amp; Jordan: <a href="http://www.cs.cmu.edu/%7Eawm/10701/readings/NgJordanNIPS2001.ps">On Discriminative and Generative Classifiers</a>, NIPS, 2001.</p>
</td>
<td></td>
</tr>
<tr>
<td>Feb 1</td>
<td></td>
<td>
<ul>
<li>Generative/Discriminative models</li>
<li>minimizing squared error and maximizing data likelihood</li>
<li>bias-variance decomposition</li>
<li>regularization</li>
</ul>
</td>
<td></td>
<td></td>
</tr>
<tr>
<td>Feb 3</td>
<td></td>
<td>
<ul>
<li>Feature selection</li>
<li>Overfitting</li>
<li>Bias-Variance tradeoff</li>
</ul>
</td>
<td></td>
<td></td>
</tr>
<tr>
<td>Feb 8</td>
<td></td>
<td>
<ul>
<li>Bayes nets</li>
<li>representing joint distributions with conditional independence assumptions</li>
</ul>
</td>
<td></td>
<td>
<p>HW3 out</p>
</td>
</tr>
<tr>
<td>Feb 15</td>
<td></td>
<td>
<ul>
<li>D-separation and Conditional Independence</li>
<li>Inference</li>
<li>Learning from fully observed data</li>
<li>Learning from partially observed data</li>
</ul>
</td>
<td></td>
<td></td>
</tr>
<tr>
<td>Feb 17</td>
<td></td>
<td>
<ul>
<li>EM</li>
</ul>
</td>
<td>
<a href="http://www.cs.cmu.edu/~aarti/Class/10701/readings/gentle_tut_HMM.pdf">EM and HMM tutorial</a> J. Bilmes</td>
<td></td>
</tr>
<tr>
<td>Feb 22</td>
<td></td>
<td>
<ul>
<li>Mixture of Gaussians clustering</li>
<li>Learning Bayes Net structure - Chow Liu</li>
</ul>
</td>
<td>
<a href="http://www.cs.cmu.edu/~aarti/Class/10701/readings/intro_gm.pdf">Intro. to Graphical Models</a>, K. Murphy<br>
<a href="http://www.cs.cmu.edu/~aarti/Class/10701/readings/graphical_model_Jordan.pdf">Graphical Models tutorial</a>, M. Jordan<br>
</td>
<td>
<p>HW3 due<br>
HW4 out</p>
</td>
</tr>
<tr>
<td>Feb 24</td>
<td></td>
<td>
<ul>
<li>PAC Learning</li>
</ul>
</td>
<td>Mitchell: Ch. 7</td>
<td></td>
</tr>
<tr>
<td>Mar 1</td>
<td></td>
<td></td>
<td></td>
<td>
<p>HW4 due</p>
</td>
</tr>
<tr>
<td>Mar 3</td>
<td>Midterm Exam</td>
<td>
<ul>
<li>in class</li>
<li>open notes, open book, no internet</li>
</ul>
</td>
<td></td>
<td>
<a href="http://www.cs.cmu.edu/~tom/10701_sp11/midterm.pdf">Midterm</a><br>
<a href="http://www.cs.cmu.edu/~tom/10701_sp11/midterm_sol.pdf">Solution</a>
</td>
</tr>
<tr>
<td>Mar 15</td>
<td></td>
<td>
<ul>
<li>Mistake bounds</li>
<li>Weighted Majority Algorithm</li>
</ul>
</td>
<td>Mitchell: Ch. 7</td>
<td></td>
</tr>
<tr>
<td>Mar 17</td>
<td></td>
<td>
<ul>
<li>CoTraining / Multi-view Learning</li>
<li>Never ending learning (NELL)</li>
</ul>
</td>
<td>
<ul>
<li><a href="http://www.cs.cmu.edu/afs/cs.cmu.edu/project/theo-20/www/mitchell-pubs/colt98_final.ps">Cotraining: Blum &amp; Mitchell</a></li>
<li><a href="http://rtw.ml.cmu.edu/papers/carlson-aaai10.pdf">NELL: Carlson et al., 2010</a></li>
</ul>
</td>
<td></td>
</tr>
<tr>
<td>Mar 22</td>
<td></td>
<td>
<ul>
<li>Markov models</li>
<li>HMM's and Bayes Nets</li>
<li>Other probabilistic time series models</li>
</ul>
</td>
<td>Bishop Ch. 13</td>
<td></td>
</tr>
<tr>
<td>Mar 24</td>
<td></td>
<td>
<ul>
<li>Non-linear regression</li>
<li>Backpropagation and Gradient descent</li>
<li>Learning hidden layer representations</li>
</ul>
</td>
<td>Mitchell Ch. 4<br>
Bishop Ch. 5</td>
<td>Project proposals due</td>
</tr>
<tr>
<td>Mar 29</td>
<td></td>
<td>
<ul>
<li>Artificial neural networks</li>
<li>PCA</li>
</ul>
</td>
<td>Bishop Ch. 12 through 12.1<br>
<a href="http://www.cs.cmu.edu/~tom/10701_sp11/slides/pca_schlens.pdf">A Tutorial on PCA</a>, J. Schlens<br>
<a href="http://www.cs.cmu.edu/~tom/10701_sp11/slides/pca_wall.pdf">SVD and PCA</a>, Wall et al.<br>
</td>
<td></td>
</tr>
<tr>
<td>Mar 31</td>
<td></td>
<td>
<ul>
<li>Deep belief networks</li>
<li>ICA</li>
<li>CCA</li>
</ul>
</td>
<td>
<a href="http://www.cs.cmu.edu/~tom/10701_sp11/slides/DeepNets_science2006.pdf">Deep Belief Nets paper</a>, Hinton &amp; Salakhutdinov<br>
<a href="http://www.cs.cmu.edu/~tom/10701_sp11/slides/CCA_tutorial.pdf">CCA Tutorial</a>, M. Borga<br>
</td>
<td></td>
</tr>
<tr>
<td>Apr 5</td>
<td></td>
<td>
<ul>
<li>Fisher Linear Discriminant</li>
<li>Latent Dirichlet Allocation</li>
<li>Intro to Kernel Functions</li>
</ul>
</td>
<td>Bishop Ch. 6.1 (required)<br>
Bishop Ch. 6.2, 6.3 (optional)</td>
<td></td>
</tr>
<tr>
<td>Apr 7</td>
<td></td>
<td>
<ul>
<li>Regression: Primal and Dual forms</li>
<li>Kernels and Kernel Regression</li>
<li>SVMs</li>
</ul>
</td>
<td>Bishop Ch. 6.1<br>
Bishop Ch. 7, through 7.1.2</td>
<td></td>
</tr>
<tr>
<td>Apr 12</td>
<td></td>
<td>
<ul>
<li>Maximizing the margin</li>
<li>Noise and soft margin SVM's</li>
<li>PAC learning and SVM's</li>
<li>Hinge loss, log loss, 0-1 loss</li>
</ul>
</td>
<td>Bishop Ch. 7, through 7.1.2</td>
<td>Project midway report due</td>
</tr>
<tr>
<td>Apr 14</td>
<td></td>
<td>No CMU classes today</td>
<td></td>
<td></td>
</tr>
<tr>
<td>Apr 19</td>
<td></td>
<td>Guest lecture: Dr. Burr Settles
<ul>
<li>Uncertainty sampling</li>
<li>Query by committee</li>
</ul>
</td>
<td>Settles: <a href="http://www.cs.cmu.edu/~bsettles/pub/settles.activelearning.pdf">Active learning survey</a>
</td>
<td></td>
</tr>
<tr>
<td>Apr 21</td>
<td></td>
<td>Guest lecture: Prof. Ziv Bar-Joseph</td>
<td></td>
<td></td>
</tr>
<tr>
<td>Apr 26</td>
<td></td>
<td>
<ul>
<li>Markov Decision Processes</li>
<li>Value Iteration</li>
<li>Q learning</li>
</ul>
</td>
<td>Kaelbling et al.: <a href="http://www.cs.cmu.edu/~tom/10701_sp11/slides/Kaelbling.pdf">Reinforcement Learning: A Survey</a>
</td>
<td></td>
</tr>
<tr>
<td>Apr 28</td>
<td></td>
<td>
<ul>
<li>Q learning in non-deterministic domains</li>
<li>RL as model for learning in animals</li>
<li>Final exam review</li>
</ul>
</td>
<td></td>
<td></td>
</tr>
<tr>
<td>May 6 (Friday)</td>
<td>Final Exam</td>
<td>
<ul>
<li>1-4pm</li>
<li>Location: Gates Hillman 4401</li>
<li>open notes, open book, no internet</li>
</ul>
</td>
<td><b><a href="http://www.cs.cmu.edu/~tom/10701_sp11/slides/FinalExamStudyTopics.pdf">Final study guide</a></b></td>
<td></td>
</tr>
</table>
</div>
</div>
